### PR TITLE
Update template and plugin references

### LINKED ISSUE #24: Authoring a template: replace library/go with protocolbuffers/go
The `Authoring a template` page uses `library/go` as the example plugin, but this plugin has been deprecated in favor of `protocolbuffers/go`. We should update the references.

### PR DESCRIPTION
**Pull request by mfridman**
Closes #24

### CODE REVIEW

File: docs/bsr/remote-generation/overview.md [hunk 1]
```diff
@@ -50,20 +44,6 @@ but shouldn't be specific to an input module. For example, you may create a temp
 JavaScript for Node.js, and one that generates JavaScript optimized for web browsers. Neither of these concepts 
 are specific to a given input module, and they could be reused by others.
 
-Buf maintains several official templates:
```
Comment (johanbrandhorst) [inline docs/bsr/remote-generation/overview.md:53]: Do we have this list somewhere else?

**Review by johanbrandhorst — APPROVED** [review]

File: docs/bsr/remote-generation/overview.md (hunk 1 above)
Comment (mfridman) [reply docs/bsr/remote-generation/overview.md:53]: There is a reference in the concepts section. You think it should be included here as well?

https://github.com/bufbuild/docs.buf.build/pull/27/files#diff-ec2711a8d6f5078aef8edbeb350147c4c7a5fac494b53932414d0f2cb9982027R51

Comment (mfridman) [reply docs/bsr/remote-generation/overview.md:53]: Most of these were just links to Protobuf plugins anyways, probably an oversight.

Comment (johanbrandhorst) [reply docs/bsr/remote-generation/overview.md:53]: I think we should make the templates more visible than that quote, like we do for the official plugins. What do you think?

Comment (mfridman) [reply docs/bsr/remote-generation/overview.md:53]: Ye that could work, let me update it.

Comment (mfridman) [reply docs/bsr/remote-generation/overview.md:53]: Updated to include this in 2 places:

Buf maintains several official templates:

- https://buf.build/grpc/templates/go
- https://buf.build/protocolbuffers/templates/go

File: docs/tour/use-remote-generation.md [hunk 2]
```diff
@@ -248,8 +248,8 @@ $ go mod tidy
 ```
 
 ```sh title="go.sum" {1-4}
--go.buf.build/library/go-grpc/$BUF_USER/petapis v1.4.4 h1:Ay1b0VFvLsey21ylibis+lP8wBiDd5RUipDnQG6nCvY=
--go.buf.build/library/go-grpc/$BUF_USER/petapis v1.4.4/go.mod h1:aKE843ItBFu7UPuaxuUJvNpqC2hjVagPYiJ20n9dBJQ=
-+go.buf.build/library/go-grpc/$BUF_USER/petapis v1.4.5 h1:kW63uI3YuRvHb4WPrn7dJQLUaMHuNE3x/912DpzwloE=
-+go.buf.build/library/go-grpc/$BUF_USER/petapis v1.4.5/go.mod h1:aKE843ItBFu7UPuaxuUJvNpqC2hjVagPYiJ20n9dBJQ=
+-go.buf.build/grpc/go/$BUF_USER/petapis v1.4.4 h1:Ay1b0VFvLsey21ylibis+lP8wBiDd5RUipDnQG6nCvY=
```
Comment (mfridman) [inline docs/tour/use-remote-generation.md:251]: The hash will be different, but I don't think it's worth updating.

**Review by johanbrandhorst — APPROVED** [review]
Thanks for this ⭐ !

**Pull request merged**